pkg/tester/data.go

```diff
@@ -154,10 +154,15 @@ func InitializeData(
 	if config.CompressionDisabled {
 		opts = append(opts, database.WithoutCompression())
 	}
+
 	if config.MemoryLimitDisabled {
+		performanceOpts := database.PerformanceBadgerOptions(dataPath)
+		// Use an extended table size for larger commits.
+		performanceOpts.MaxTableSize = config.MaxTableSize
```

Comment:
I'd conditionally overwrite `performanceOpts.MaxTableSize` if `config.MaxTableSize` is not `nil`. (once you make the pointer change above)